cf-testd/cf-testd.c

```diff
@@ -632,8 +633,32 @@ static void *CFTestD_ServeReport(void *config_arg)
             Log(LOG_LEVEL_ERR, "Error reading report file '%s'", report_file);
             exit(EXIT_FAILURE);
         }
-        config->report     = StringWriterClose(contents);
-        config->report_len = SafeStringLength(config->report);
+
+        size_t report_data_len = StringWriterLength(contents);
+        char *report_data = StringWriterClose(contents);
+
```

Comment:
Also wondering how StringWriterClose() works... seems like we are returned `char *data = writer->string.data` and that function does a `free(writer)` so wouldn't the memory pointed to by `return data` be freed?


```
char *StringWriterClose(Writer *writer)
// NOTE: transfer of ownership for allocated return value
{
    if (writer->type != WT_STRING)
    {
        ProgrammingError("Wrong writer type");
    }
    char *data = writer->string.data;

    free(writer);
    return data;
}
```

Do we need to free `char *report_data`?

Reply:
@craigcomstock yes, StringWriterClose gives you "ownership" of the string, so you are correct in saying that report_data needs to be freed.